lib/fluent/plugin/in_mysql_replicator.rb

```diff
@@ -133,8 +138,9 @@ def emit_record(tag, record)
 
     def query(query, con = nil)
       begin
-        mysql = get_connection if con.nil?
-        return mysql.query(query)
+        con = con.nil? ? get_connection : con
+        con = con.ping ? con : get_connection
```

Comment:
>Is this the solution to reconnect on lost connection in the loop?
Yes.
If Mysql server is down,  prepared_con.query is Error and shutdown.
And Not reconnect.(re-use connection)